C error when doing a fancy sort

```
$ v doctor
OS: linux, Debian GNU/Linux 9.13 (stretch) (VM)
Processor: 2 cpus, 64bit, little endian, Intel(R) Core(TM) i7-7820HQ CPU @ 2.90GHz
CC version: cc (Debian 6.3.0-18+deb9u1) 6.3.0 20170516

getwd: /home/user/gocode/src/github.com/elimisteve/design-academy-ventures-site/db
vmodules: /home/user/.vmodules
vroot: /home/user/django_projects/v
vexe: /home/user/django_projects/v/v
vexe mtime: 2021-09-01 05:26:08
is vroot writable: true
is vmodules writable: true
V full version: V 0.2.4 900c37a.56993b9

Git version: git version 2.11.0
Git vroot status: weekly.2021.25-713-g56993b9e
.git/config present: true
thirdparty/tcc status: thirdparty-linux-amd64 333c14de
```

**What did you do?**

```bash
$ cat files2json-err.v
import os

fn main() {
        text := os.read_file(os.args[0]) ?
        mut lines := text.split_into_lines()
        lines.sort(a.split('/').last() < b.split('/').last())
        println(lines.join('\n'))
}

$ v run files2json-err.v
```


**What did you expect to see?**

The lines of the current file listed in a different order
 
**What did you see instead?**

C error:

```
$ v -cg run files2json-err.v
/tmp/v/files2json-err.7559863757417673810.tmp.c:2410: error: identifier expected
builder error: 
==================
C error. This should never happen.

This is a compiler bug, please report it using `v bug file.v`.

https://github.com/vlang/v/issues/new/choose

You can also use #help on Discord: https://discord.gg/vlang
```

Culprit is probably this mangled-looking C line:

> int compare_string_by_split('/')_last()(string* a, string* b) {

--- a/vlib/v/checker/checker.v
+++ b/vlib/v/checker/checker.v
@@ -2555,6 +2555,15 @@ fn (mut c Checker) array_builtin_method_call(mut node ast.CallExpr, left_type as
 				} else if left_name == right_name {
 					c.error('`.sort()` cannot use same argument', node.pos)
 				}
+				if (node.args[0].expr.left !is ast.Ident
+					&& node.args[0].expr.left !is ast.SelectorExpr
+					&& node.args[0].expr.left !is ast.IndexExpr)
+					|| (node.args[0].expr.right !is ast.Ident
+					&& node.args[0].expr.right !is ast.SelectorExpr
+					&& node.args[0].expr.right !is ast.IndexExpr) {
+					c.error('`.sort()` can only use ident, index or selector as argument, \ne.g. `arr.sort(a < b)`, `arr.sort(a.id < b.id)`, `arr.sort(a[0] < b[0])`',
+						node.pos)
+				}
 			} else {
 				c.error(
 					'`.sort()` requires a `<` or `>` comparison as the first and only argument' +

--- a/vlib/v/checker/tests/array_fancy_sort_err.out
+++ b/vlib/v/checker/tests/array_fancy_sort_err.out
@@ -1,0 +1,8 @@
+vlib/v/checker/tests/array_fancy_sort_err.vv:6:8: error: `.sort()` can only use ident, index or selector as argument,
+e.g. `arr.sort(a < b)`, `arr.sort(a.id < b.id)`, `arr.sort(a[0] < b[0])`
+    4 |     text := os.read_file(os.args[0]) ?
+    5 |     mut lines := text.split_into_lines()
+    6 |     lines.sort(a.split('/').last() < b.split('/').last())
+      |           ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
+    7 |     println(lines.join('\n'))
+    8 | }

--- a/vlib/v/checker/tests/array_fancy_sort_err.vv
+++ b/vlib/v/checker/tests/array_fancy_sort_err.vv
@@ -1,0 +1,8 @@
+import os
+
+fn main() {
+	text := os.read_file(os.args[0]) ?
+	mut lines := text.split_into_lines()
+	lines.sort(a.split('/').last() < b.split('/').last())
+	println(lines.join('\n'))
+}

